resharper/resharper-unity/src/CSharp/Feature/Respeller/ReSharper/UnityInstalledDictionarySettingsProvider.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using JetBrains.Annotations;
+using JetBrains.Application.Settings;
+using JetBrains.Application.Settings.Implementation;
+using JetBrains.ProjectModel;
+using JetBrains.ReSharper.Plugins.Unity.Settings;
+using JetBrains.Util;
+
+namespace JetBrains.ReSharper.Plugins.Unity.CSharp.Feature.Respeller
+{
+    [SolutionComponent]
+    public class UnityInstalledDictionarySettingsProvider : IUnitySolutionSettingsProvider
+    {
+        private readonly ISettingsSchema mySettingsSchema;
+        private readonly ILogger myLogger;
+
+        public UnityInstalledDictionarySettingsProvider(ISettingsSchema settingsSchema, ILogger logger)
+        {
+            mySettingsSchema = settingsSchema;
+            myLogger = logger;
+        }
+
+        public void InitialiseSolutionSettings(ISettingsStorageMountPoint mountPoint)
+        {
+            var entry = GetIndexedEntry();
+            if (entry == null)
+                return;
+            var path = (FileSystemPath.Parse(GetType().Assembly.Location).Parent / "Extensions/JetBrains.unity/dictionaries/unity.dic").FullPath;
+            ScalarSettingsStoreAccess.SetIndexedValue(mountPoint, entry, path, null, true, null, myLogger);
+        }
+
+        [CanBeNull]
+        private SettingsIndexedEntry GetIndexedEntry()
+        {
+            // TODO: Fix reference to Psi.Features.VisualStudio and use the InstalledDictionariesSettings type directly
+            var type = Type.GetType(
+                "JetBrains.PsiFeatures.VisualStudio.ReSpeller.InstalledDictionariesSettings, " +
+                "JetBrains.PsiFeatures.VisualStudio.ReSpeller, Version=777.0.0.0, Culture=neutral, PublicKeyToken=1010a0d8d6380325");
+            if (type == null)
```

Comment:
Using our own dictionary will have a perf impact, because `IDictionarySource` is a viewable set, and will call `Daemon.Invalidate` for each item in the set, and I'm adding about 250 items.

The reflection is only because I don't know how to reference the Psi.Features.VisualStudio package after @mfilippov refactored to use the Rider SDK, rather than the ReSharper SDK.

Reply:
hmm. got it. didn't know about this SDK problem.

Reply:
Seems I need to add some API to support embedded dictionaries

Reply:
What does that mean for this PR? Will you add a new API for 201?

Reply:
Not sure we need a new API now, as I can add a dictionary cleanly now that we have proper SDK referencing. No dirty hacks any more, just adding a dictionary into an in-memory settings layer for Unity projects.